spdmlib/src/common/mod.rs

```diff
@@ -182,9 +186,22 @@ impl<'a> SpdmContext<'a> {
             .append_message(self.runtime_info.message_a.as_ref())
             .ok_or(spdm_err!(ENOMEM))?;
         debug!("message_a - {:02x?}", self.runtime_info.message_a.as_ref());
+
+        if self.peer_info.peer_cert_chain[slot_id as usize].is_none() && !use_psk {
```

Comment:
I think this check should be moved into !use_psk

Reply:
> I think this check should be moved into !use_psk

Updated.